utils/rowgroup/rowaggregation.cpp

```diff
@@ -751,9 +680,47 @@ void RowAggregation::initialize()
 {
     // Calculate the length of the hashmap key.
     fAggMapKeyCount = fGroupByCols.size();
+    bool disk_agg = fRm ? fRm->getAllowDiskAggregation() : false;
+    bool allow_gen = true;
+    for (auto& fun : fFunctionCols)
+    {
+      if (fun->fAggFunction == ROWAGG_UDAF || fun->fAggFunction == ROWAGG_GROUP_CONCAT)
+      {
+        allow_gen = false;
+        break;
+      }
+    }
+
+    config::Config* config = config::Config::makeConfig();
+    string tmpDir = config->getConfig("RowAggregation", "TempDir");
+    if (tmpDir.empty() || tmpDir == "/")
+      tmpDir.assign("/tmp/cs-agg");
```

Comment:
A questionable default since we want to offload the data from RAM and /tmp in most cases is tmpfs. I think it would be better to check if disk-based aggregation is enable and check if RowAggregation.TempDir is set. If not we should throw an error.